cluster/src/main/scala/com/linkedin/norbert/cluster/Node.scala

```diff
@@ -42,9 +42,9 @@ object Node {
       val partitions = node.getPartitionList.asInstanceOf[java.util.List[Int]].foldLeft(Set[Int]()) { (set, i) => set + i }
 
       if(!node.hasPersistentCapability)
```

Comment:
With this many if else cases, it might be worthwhile to individually calculate the arguments e.g.
val persistentCapability = if (node.hasPersistenCapability) Some(...) else None
val offset = if (...) Some(...) else None